pkg/server/conn.go

```diff
@@ -2581,3 +2599,41 @@ func (cc *clientConn) ReadPacket() ([]byte, error) {
 func (cc *clientConn) Flush(ctx context.Context) error {
 	return cc.flush(ctx)
 }
+
+// Stats returns the connection statistics.
+func (*clientConn) Stats(vars *variable.SessionVars) (map[string]interface{}, error) {
+	m := make(map[string]interface{}, len(clientConnStatus))
+	for name, v := range clientConnStatus {
```

Comment:
I'm not very sure about the usage of `clientConnStatus`. After `variable.RegisterStatistics(cc)` the scope part of  `clientConnStatus` is not used, can we remove `clientConnStatus`?

Reply:
Not sure exactly what you mean with this. Could you explain it a bit more?

Reply:
I think we can remove `clientConnStatus`.

```diff
diff --git a/pkg/server/conn.go b/pkg/server/conn.go
index 4518be086b..58a9e24942 100644
--- a/pkg/server/conn.go
+++ b/pkg/server/conn.go
@@ -120,12 +120,6 @@ var (
        statusCompressionLevel     = "Compression_level"
 )

-var clientConnStatus = map[string]*variable.StatusVal{
-       statusCompression:          {Scope: variable.ScopeSession, Value: "OFF"},
-       statusCompressionAlgorithm: {Scope: variable.ScopeSession, Value: ""},
-       statusCompressionLevel:     {Scope: variable.ScopeSession, Value: 0},
-}
-
 // newClientConn creates a *clientConn object.
 func newClientConn(s *Server) *clientConn {
        cc := &clientConn{
@@ -2602,10 +2596,7 @@ func (cc *clientConn) Flush(ctx context.Context) error {

 // Stats returns the connection statistics.
 func (*clientConn) Stats(vars *variable.SessionVars) (map[string]interface{}, error) {
-       m := make(map[string]interface{}, len(clientConnStatus))
-       for name, v := range clientConnStatus {
-               m[name] = v.Value
-       }
+       m := make(map[string]interface{}, 3)

        switch vars.CompressionAlgorithm {
        case mysql.CompressionNone:
```

Reply:
Done in 6081df6cf805cb3df553e26471d47ccc714bbb0f